### DISCUSSION TITLE
Quick one: nicer way to do this?

### DISCUSSION THREAD

**Discussion by pipedreambomb** [ramda/ramda#3358] [Q&A] [answered]
This is mostly just for fun or for cleaner code, but I have something that looks funky:

```
    color = cond([
      ...
      [c => colorMap[c.value], c => colorMap[c.value]],
      ...
    ])
```
As you can see, I have the same thing on both sides of a `cond` transformation thingy, because both my predicate and transformer are the same. If it finds a color in my `colorMap` (ie the result of accessing the array isn't null or `undefined`), then it should return it. 

I just wondered if I'd missed a nicer way to put this that doesn't do the same computation twice. Maybe this is just how `cond` has to be, though.

**Comment by mendrik** [comment]
why would you need cond() for this? a simple  `colorMap[c.value]` does exactly what you want, if map lacks the color, it will be undefined.

if you need some default you could even just do propOr(defaultColor, c.value, colorMap)

**Comment by pipedreambomb** [comment]
I missed out the rest of the conditions. That's what the ... represent.

**Comment by kedashoe** [comment, marked as answer]
you could just update the first part
```js
color = cond([
      ...
      [c => c.value in colorMap, c => colorMap[c.value]],
      ...
    ])
```
or more ramda-y
```js
color = cond([
      ...
      [pipe(prop('value'), has(__, colorMap)), pipe(prop('value'), prop(__, colorMap))],
      ...
    ])
```

**Comment by pipedreambomb** [comment]
Weirdly, I've been doing JavaScript for years and I don't think I've ever
used the keyword `in` except for inside a `for` loop. I like it, thanks,
I'll use that!
